pkgs/development/libraries/qt-5/modules/qtbase.nix

```diff
@@ -376,6 +376,18 @@ stdenv.mkDerivation {
       moveToOutput bin "$dev"
     ''
 
+    # Fix paths returned by qmake -query
+    + ''
+      cat > $dev/bin/qt.conf <<EOF
```

Comment:
What will happen if the `dev` output is installed in a user environment?

Reply:
I am not sure I know all the effects of a derivation being installed in a user environment as I am by far not a Nix expert, but as long as that only entails the ```bin``` subfolder being added to the ```PATH```, the added file should have no side effects, at least according official [documentation on qt.conf](https://doc.qt.io/qt-5/qt-conf.html):

    QLibraryInfo will load qt.conf from one of the following locations:

    - :/qt/etc/qt.conf using the resource system
    - on macOS, in the Resource directory inside the application bundle, for example assistant.app/Contents/Resources/qt.conf
    - in the directory containing the application executable, i.e. QCoreApplication::applicationDirPath() + QDir::separator() + "qt.conf"

The ```qt.conf``` file should only really affect the binaries inside the derivations ```bin``` folder

Reply:
> The `qt.conf` file should only really affect the binaries inside the derivations `bin` folder

Installing it into a user environment will cause it to be symlinked into another directory with (symlinks to) other programs. I'm concerned that this will break other Qt programs if `QCoreApplication::applicationDirPath` is based on `argv[0]`. I will check...

Reply:
It looks like `QCoreApplication::applicationDirPath` [gets the program filename from `/proc`](https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/kernel/qcoreapplication.cpp#n2269), which is fine, but it does fall back to `argv[0]`. So, I think this change should be fine.